h2/stream.py

```diff
@@ -206,6 +216,9 @@ def stream_ended(self, previous_state):
         """
         Fires when a stream is cleanly ended.
         """
+        assert previous_state is StreamState.HALF_CLOSED_REMOTE
```

Comment:
Better to use == than is, as we only care about equality of these states, not identity.